examples/language-sdk-instrumentation/golang-push/rideshare-k6/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"rideshare/bike"
+	"rideshare/car"
+	"rideshare/rideshare"
+	"rideshare/scooter"
+	"rideshare/utility"
+
+	otellogs "github.com/agoda-com/opentelemetry-logs-go"
+	otelpyroscope "github.com/grafana/otel-profiling-go"
+	"github.com/grafana/pyroscope-go/x/k6"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+func bikeRoute(w http.ResponseWriter, r *http.Request) {
+	bike.OrderBike(r.Context(), 1)
+	w.Write([]byte("<h1>Bike ordered</h1>"))
+}
+
+func scooterRoute(w http.ResponseWriter, r *http.Request) {
+	scooter.OrderScooter(r.Context(), 2)
+	w.Write([]byte("<h1>Scooter ordered</h1>"))
+}
+
+func carRoute(w http.ResponseWriter, r *http.Request) {
+	car.OrderCar(r.Context(), 3)
+	w.Write([]byte("<h1>Car ordered</h1>"))
+}
+
+func index(w http.ResponseWriter, r *http.Request) {
+	rideshare.Log.Print(r.Context(), "showing index")
+	result := "<h1>environment vars:</h1>"
+	for _, env := range os.Environ() {
+		result += env + "<br>"
+	}
+	w.Write([]byte(result))
+}
+
+func main() {
+	config := rideshare.ReadConfig()
+
+	tp, _ := setupOTEL(config)
+	defer func() {
+		_ = tp.Shutdown(context.Background())
+	}()
+
+	p, err := rideshare.Profiler(config)
+
+	if err != nil {
+		log.Fatalf("error starting pyroscope profiler: %v", err)
+	}
+	defer func() {
+		_ = p.Stop()
+	}()
+
+	cleanup := utility.InitWorkerPool(config)
+	defer cleanup()
+
+	rideshare.Log.Print(context.Background(), "started ride-sharing app")
+
+	http.Handle("/", k6.LabelsFromBaggageHandler(otelhttp.NewHandler(http.HandlerFunc(index), "IndexHandler")))
+	http.Handle("/bike", k6.LabelsFromBaggageHandler(otelhttp.NewHandler(http.HandlerFunc(bikeRoute), "BikeHandler")))
+	http.Handle("/scooter", k6.LabelsFromBaggageHandler(otelhttp.NewHandler(http.HandlerFunc(scooterRoute), "ScooterHandler")))
+	http.Handle("/car", k6.LabelsFromBaggageHandler(otelhttp.NewHandler(http.HandlerFunc(carRoute), "CarHandler")))
```

Comment:
Here is where we wrap the handlers with the k6 middleware.

Reply:
Maybe adding it as a comment?
```golang
// The handlers are wrapped with k6 middleware to provide profiling information during k6 load tests.
```

Reply:
@marcsanmi Good point! In https://github.com/grafana/pyroscope/pull/3648/commits/0957f28f861724614491f17fbd0dd1a2256263ce and https://github.com/grafana/pyroscope/pull/3648/commits/9e3b2e7952a0792a0ae25ba7519fbc1cfdd4b7e7 I refactored how routes are registered and added a comment on the purpose of `applyK6Middleware`.